b2/sync.py

```diff
@@ -537,6 +537,12 @@ def _walk_relative_paths(self, prefix_len, dir_path, reporter):
         names = {}  # name to (full_path, relative path)
         dirs = set()  # subset of names that are directories
         for name in os.listdir(dir_path):
+            # We expect listdir() to return unicode if dir_path is unicode.
+            # This assertion is to help track down issue #190.  One theory
+            # is that listdir() is not behaving correctly.  If it were to
+            # return bytes rather than unicode, it's not clear how to decode it.
+            assert isinstance(name, six.text_type)
```

Comment:
if `os.listdir` behaves incorrectly, we'll crash the entire program (in the major release). Could we produce a warning instead, to keep the debug help, but not crash?